docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,403 @@
+
+## When to Use `simulateTransaction` (whenever you use soroban :D)
+
+1. **Preparing `invokeHostFunctionOp` Transactions**: Anytime you need to submit an `invokeHostFunctionOp` transaction to the network.
+2. **Footprint Determination**: To determine the ledger footprint, which includes all the data entries the transaction will read or write.
+3. **Authorization Identification**: To identify the authorizations required for the transaction.
+4. **Error Detection**: To detect potential errors and issues before actual submission, saving time and network resources.
+5. **Restoring Archived Ledger Entries or Contract Code**: To prepare and restore archived data before actual transaction submission.
+6. **Simulating Contract Getter Calls**: To retrieve certain data from the contract without affecting the ledger state.
+7. **Resource Calculation**: To calculate the necessary resources (CPU instructions, memory, etc.) that a transaction will consume.
+
+## How to Call `simulateTransaction`
+
+### Using Fetch
+
+Here's an example of how to call the `simulateTransaction` endpoint directly using `fetch` in JavaScript:
+
+```javascript
+async function simulateTransaction(transactionXDR) {
+  const requestBody = {
+    jsonrpc: "2.0",
+    id: 8675309,
+    method: "simulateTransaction",
+    params: {
+      transaction: transactionXDR,
+      resourceConfig: {
+        instructionLeeway: 50000,
+      },
+    },
+  };
+
+  const response = await fetch("https://soroban-testnet.stellar.org:443", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(requestBody),
+  });
+
+  const result = await response.json();
+  console.log(result);
+}
+
+// Example XDR transaction envelope
+// Replace the following placeholder with your actual XDR transaction envelope
+const transactionXDR =
+  "AAAAAgAAAAAg4dbAxsGAGICfBG3iT2cKGYQ6hK4sJWzZ6or1C5v6GAAAAGQAJsOiAAAAEQAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAGAAAAAAAAAABzAP+dP0PsNzYvFF1pv7a8RQXwH5eg3uZBbbWjE9PwAsAAAAJaW5jcmVtZW50AAAAAAAAAgAAABIAAAAAAAAAACDh1sDGwYAYgJ8EbeJPZwoZhDqEriwlbNnqivULm/oYAAAAAwAAAAMAAAAAAAAAAAAAAAA=";
+simulateTransaction(transactionXDR);
+```
+
+### Using the JavaScript SDK
+
+The Stellar SDK provides a convenient method to simulate a transaction:
+
+```javascript
+const StellarSdk = require("stellar-sdk");
+const { Keypair, SorobanRpc } = StellarSdk;
+
+const rpc_url = 'https://soroban-testnet.stellar.org:443'
+
+// Generate a new keypair for transaction authorization.
+const keypair = Keypair.random();
+const secret = keypair.secret();
+const public = keypair.publicKey();
+
+// Initialize the rpcServer
+const rpcServer = new SorobanRpc.Server(rpc_url, { allowHttp: true })
+
+// Load the account (getting the sequence number for the account and making an account object.)
+Const account = rpcServer.getAccount(public)
```

Comment:
```suggestion
const account = rpcServer.getAccount(public)
```

Reply:
@Shaptic, could you please take a look at this script and adjust it to show simulated transaction results? Your expertise with the SDK would be greatly appreciated.